libs/game/multiplayer.ts

```diff
@@ -25,14 +25,53 @@ namespace multiplayer {
         game.pushScene();
     }
 
+    const MULTIPLAYER_PLAYER_JOINED_ID = 3241;
+    const MULTIPLAYER_PLAYER_LEFT_ID = 3242;
     export function initServer() {
         if (getOrigin() === "server") {
             game.eventContext().registerFrameHandler(scene.MULTIPLAYER_POST_SCREEN_PRIORITY, () => {
                 if (getOrigin() === "server") {
                     postImage(screen);
                 }
             })
+
+            for (let p = 1; p <= 4; p++) {
```

Comment:
4 is hard-coded in so many places...

Reply:
Yeahh, we should probably do a pass and add a multiplayer.MAX_PLAYERS or something.

And for the other point, should just be adding a helper block for reading out https://github.com/microsoft/pxt-common-packages/blob/af1990f29230881ea56822fa9b5323d02334fde9/libs/game/controller.ts#L227